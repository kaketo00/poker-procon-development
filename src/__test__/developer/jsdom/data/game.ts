import type { Game } from '@/schema/game';

export const testGameInfoData: Game = {
  id: '10',
  status: 'progress',
  createdAt: '2024-04-15T18:52:10.296Z',
  players: {
    Player1: {
      name: 'Player1',
      status: 'active',
      point: 37600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [
          {
            suit: 'Clubs',
            number: 1,
          },
          {
            suit: 'Diamonds',
            number: 7,
          },
          {
            suit: 'Spades',
            number: 6,
          },
          {
            suit: 'Spades',
            number: 9,
          },
          {
            suit: 'Clubs',
            number: 8,
          },
        ],
        hand: 1,
      },
    },
    Player2: {
      name: 'Player2',
      status: 'active',
      point: 17600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [
          {
            suit: 'Hearts',
            number: 13,
          },
          {
            suit: 'Hearts',
            number: 1,
          },
          {
            suit: 'Spades',
            number: 5,
          },
          {
            suit: 'Clubs',
            number: 4,
          },
          {
            suit: 'Hearts',
            number: 4,
          },
        ],
        hand: 2,
      },
    },
    Player3: {
      name: 'Player3',
      status: 'active',
      point: 24800,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [
          {
            suit: 'Diamonds',
            number: 12,
          },
          {
            suit: 'Diamonds',
            number: 5,
          },
          {
            suit: 'Diamonds',
            number: 2,
          },
          {
            suit: 'Spades',
            number: 1,
          },
          {
            suit: 'Diamonds',
            number: 1,
          },
        ],
        hand: 2,
      },
    },
    Player4: {
      name: 'Player4',
      status: 'out',
      point: 0,
      round: {
        betPoint: 0,
        action: null,
        cards: [],
      },
    },
  },
  totalRound: 100,
  initialPoint: 20000,
  fee: 200,
  seatingOrder: ['Player3', 'Player4', 'Player1', 'Player2'],
  startedAt: '2024-04-15T18:52:13.832Z',
};
