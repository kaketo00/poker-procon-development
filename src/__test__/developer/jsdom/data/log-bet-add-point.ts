import type { Log } from '@/schema/log';

export const testLogBetAddPoint: Log = {
  round: 10,
  player: 'Player1',
  phase: 'bet-1',
  content: { type: 'bet', amount: 314, action: 'bet', payPoint: 314 },
  game: {
    id: '10',
    status: 'progress',
    createdAt: '2024-04-22T12:06:55.706Z',
    players: {
      Player1: {
        name: 'Player1',
        status: 'active',
        point: 64746,
        round: {
          betPoint: 314,
          first: 314,
          second: 0,
          action: 'bet',
          cards: [
            { suit: 'Clubs', number: 6 },
            { suit: 'Spades', number: 8 },
            { suit: 'Hearts', number: 9 },
            { suit: 'Diamonds', number: 1 },
            { suit: 'Spades', number: 1 },
          ],
        },
      },
      Player2: {
        name: 'Player2',
        status: 'active',
        point: 6256,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: null,
          cards: [
            { suit: 'Diamonds', number: 6 },
            { suit: 'Clubs', number: 7 },
            { suit: 'Diamonds', number: 7 },
            { suit: 'Diamonds', number: 12 },
            { suit: 'Clubs', number: 1 },
          ],
        },
      },
      Player3: {
        name: 'Player3',
        status: 'out',
        point: 0,
        round: { betPoint: 0, first: 0, second: 0, action: null, cards: [] },
      },
      Player4: {
        name: 'Player4',
        status: 'active',
        point: 8084,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: null,
          cards: [
            { suit: 'Diamonds', number: 3 },
            { suit: 'Hearts', number: 4 },
            { suit: 'Diamonds', number: 4 },
            { suit: 'Spades', number: 11 },
            { suit: 'Hearts', number: 11 },
          ],
        },
      },
    },
    totalRound: 100,
    initialPoint: 20000,
    fee: 200,
    seatingOrder: ['Player2', 'Player1', 'Player4', 'Player3'],
    startedAt: '2024-04-22T12:06:56.923Z',
  },
  pot: 914,
  minBetPoint: 314,
  createdAt: 1713787617161,
};
