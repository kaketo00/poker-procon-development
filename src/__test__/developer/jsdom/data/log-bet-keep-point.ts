import type { Log } from '@/schema/log';

export const testLogBetKeepPoint: Log = {
  round: 10,
  player: 'Player1',
  phase: 'bet-1',
  content: { type: 'bet', amount: -1, action: 'drop', payPoint: 0 },
  game: {
    id: '10',
    status: 'progress',
    createdAt: '2024-04-22T12:06:55.706Z',
    players: {
      Player1: {
        name: 'Player1',
        status: 'active',
        point: 64746,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: 'drop',
          cards: [],
        },
      },
      Player2: {
        name: 'Player2',
        status: 'active',
        point: 6256,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: null,
          cards: [
            { suit: 'Diamonds', number: 6 },
            { suit: 'Clubs', number: 7 },
            { suit: 'Diamonds', number: 7 },
            { suit: 'Diamonds', number: 12 },
            { suit: 'Clubs', number: 1 },
          ],
        },
      },
      Player3: {
        name: 'Player3',
        status: 'out',
        point: 0,
        round: { betPoint: 0, first: 0, second: 0, action: null, cards: [] },
      },
      Player4: {
        name: 'Player4',
        status: 'active',
        point: 8084,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: null,
          cards: [
            { suit: 'Diamonds', number: 3 },
            { suit: 'Hearts', number: 4 },
            { suit: 'Diamonds', number: 4 },
            { suit: 'Spades', number: 11 },
            { suit: 'Hearts', number: 11 },
          ],
        },
      },
    },
    totalRound: 100,
    initialPoint: 20000,
    fee: 200,
    seatingOrder: ['Player2', 'Player1', 'Player4', 'Player3'],
    startedAt: '2024-04-22T12:06:56.923Z',
  },
  pot: 600,
  minBetPoint: 0,
  createdAt: 1713787617161,
};
