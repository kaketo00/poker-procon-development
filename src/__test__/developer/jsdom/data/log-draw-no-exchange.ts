import type { Log } from '@/schema/log';

export const testLogDrawNoExchange: Log = {
  round: 10,
  player: 'Player1',
  phase: 'draw-1',
  content: {
    type: 'draw',
    exchange: [false, false, false, false, false],
    keep: [
      { suit: 'Spades', number: 8 },
      { suit: 'Hearts', number: 9 },
      { suit: 'Diamonds', number: 1 },
      { suit: 'Spades', number: 1 },
      { suit: 'Hearts', number: 1 },
    ],
    discard: [],
    newCard: [],
    before: [
      { suit: 'Spades', number: 8 },
      { suit: 'Hearts', number: 9 },
      { suit: 'Diamonds', number: 1 },
      { suit: 'Spades', number: 1 },
      { suit: 'Hearts', number: 1 },
    ],
    after: [
      { suit: 'Spades', number: 8 },
      { suit: 'Hearts', number: 9 },
      { suit: 'Diamonds', number: 1 },
      { suit: 'Spades', number: 1 },
      { suit: 'Hearts', number: 1 },
    ],
  },
  game: {
    id: '10',
    status: 'progress',
    createdAt: '2024-04-22T12:06:55.706Z',
    players: {
      Player1: {
        name: 'Player1',
        status: 'active',
        point: 64746,
        round: {
          betPoint: 314,
          action: 'bet',
          cards: [
            { suit: 'Spades', number: 8 },
            { suit: 'Hearts', number: 9 },
            { suit: 'Diamonds', number: 1 },
            { suit: 'Spades', number: 1 },
            { suit: 'Hearts', number: 1 },
          ],
        },
      },
      Player2: {
        name: 'Player2',
        status: 'active',
        point: 5942,
        round: {
          betPoint: 314,
          action: 'call',
          cards: [
            { suit: 'Diamonds', number: 6 },
            { suit: 'Clubs', number: 7 },
            { suit: 'Diamonds', number: 7 },
            { suit: 'Diamonds', number: 12 },
            { suit: 'Clubs', number: 1 },
          ],
        },
      },
      Player3: {
        name: 'Player3',
        status: 'out',
        point: 0,
        round: { betPoint: 0, action: null, cards: [] },
      },
      Player4: {
        name: 'Player4',
        status: 'active',
        point: 7770,
        round: {
          betPoint: 314,
          action: 'call',
          cards: [
            { suit: 'Diamonds', number: 3 },
            { suit: 'Hearts', number: 4 },
            { suit: 'Diamonds', number: 4 },
            { suit: 'Spades', number: 11 },
            { suit: 'Hearts', number: 11 },
          ],
        },
      },
    },
    totalRound: 100,
    initialPoint: 20000,
    fee: 200,
    seatingOrder: ['Player2', 'Player1', 'Player4', 'Player3'],
    startedAt: '2024-04-22T12:06:56.923Z',
  },
  pot: 1542,
  minBetPoint: 314,
  createdAt: 1713787617162,
};
