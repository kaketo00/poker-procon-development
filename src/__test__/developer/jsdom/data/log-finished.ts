import type { Log } from '@/schema/log';

export const testLogFinished: Log = {
  round: 10,
  player: '',
  phase: 'finished',
  content: {
    type: 'finished',
    pot: 7461,
    winner: 'Player1',
    result: {
      Player1: {
        cards: [
          { suit: 'Spades', number: 4 },
          { suit: 'Clubs', number: 4 },
          { suit: 'Clubs', number: 5 },
          { suit: 'Diamonds', number: 8 },
          { suit: 'Diamonds', number: 13 },
        ],
        hand: 2,
      },
      Player4: {
        cards: [
          { suit: 'Hearts', number: 2 },
          { suit: 'Clubs', number: 3 },
          { suit: 'Hearts', number: 8 },
          { suit: 'Clubs', number: 10 },
          { suit: 'Spades', number: 11 },
        ],
        hand: 1,
      },
      Player3: { cards: [], hand: 0 },
      Player2: {
        cards: [
          { suit: 'Clubs', number: 2 },
          { suit: 'Spades', number: 3 },
          { suit: 'Diamonds', number: 5 },
          { suit: 'Diamonds', number: 10 },
          { suit: 'Clubs', number: 12 },
        ],
        hand: 1,
      },
    },
  },
  game: {
    id: '10',
    status: 'progress',
    createdAt: '2024-04-22T12:06:55.706Z',
    players: {
      Player1: {
        name: 'Player1',
        status: 'active',
        point: 70234,
        round: {
          betPoint: 2287,
          action: 'call',
          cards: [
            { suit: 'Spades', number: 4 },
            { suit: 'Clubs', number: 4 },
            { suit: 'Clubs', number: 5 },
            { suit: 'Diamonds', number: 8 },
            { suit: 'Diamonds', number: 13 },
          ],
          hand: 2,
        },
      },
      Player2: {
        name: 'Player2',
        status: 'active',
        point: 3969,
        round: {
          betPoint: 2287,
          action: 'call',
          cards: [
            { suit: 'Clubs', number: 2 },
            { suit: 'Spades', number: 3 },
            { suit: 'Diamonds', number: 5 },
            { suit: 'Diamonds', number: 10 },
            { suit: 'Clubs', number: 12 },
          ],
          hand: 1,
        },
      },
      Player3: {
        name: 'Player3',
        status: 'out',
        point: 0,
        round: { betPoint: 0, action: null, cards: [] },
      },
      Player4: {
        name: 'Player4',
        status: 'active',
        point: 5797,
        round: {
          betPoint: 2287,
          action: 'raise',
          cards: [
            { suit: 'Hearts', number: 2 },
            { suit: 'Clubs', number: 3 },
            { suit: 'Hearts', number: 8 },
            { suit: 'Clubs', number: 10 },
            { suit: 'Spades', number: 11 },
          ],
          hand: 1,
        },
      },
    },
    totalRound: 100,
    initialPoint: 20000,
    fee: 200,
    seatingOrder: ['Player2', 'Player1', 'Player4', 'Player3'],
    startedAt: '2024-04-22T12:06:56.923Z',
  },
  pot: 7461,
  minBetPoint: 2287,
  createdAt: 1713787617170,
};
