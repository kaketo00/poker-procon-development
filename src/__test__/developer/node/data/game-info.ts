import type { GameInfo } from '@/schema/game';

export const testGameInfoData: GameInfo = {
  currentRound: 1,
  phase: 'start',
  order: ['Player1', 'Player2', 'Player3', 'Player4'],
  pot: 0,
  minBetPoint: 0,
  players: {
    Player1: {
      name: 'Player1',
      status: 'active',
      point: 37600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    Player2: {
      name: 'Player2',
      status: 'active',
      point: 17600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    Player3: {
      name: 'Player3',
      status: 'active',
      point: 24800,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    Player4: {
      name: 'Player4',
      status: 'out',
      point: 0,
      round: {
        betPoint: 0,
        action: null,
        cards: [],
      },
    },
  },
  totalRound: 100,
  initialPoint: 20000,
  fee: 200,
};
