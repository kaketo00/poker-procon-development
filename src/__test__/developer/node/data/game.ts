import type { Game } from '@/schema/game';

export const testGameData: Game = {
  id: '1',
  status: 'new',
  createdAt: '2024-05-01T10:15:19.019Z',
  players: {
    DemoPlayer1: {
      name: 'DemoPlayer1',
      status: 'active',
      point: 37600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    DemoPlayer2: {
      name: 'DemoPlayer2',
      status: 'active',
      point: 17600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    DemoPlayer3: {
      name: 'DemoPlayer3',
      status: 'active',
      point: 24800,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    DemoPlayer4: {
      name: 'DemoPlayer4',
      status: 'out',
      point: 0,
      round: {
        betPoint: 0,
        action: null,
        cards: [],
      },
    },
  },
  totalRound: 100,
  initialPoint: 20000,
  fee: 200,
  seatingOrder: ['DemoPlayer1', 'DemoPlayer2', 'DemoPlayer3', 'DemoPlayer4'],
  startedAt: '2024-05-01T10:52:16.257Z',
};
