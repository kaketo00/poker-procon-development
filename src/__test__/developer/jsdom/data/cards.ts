import type { Card, CardSet } from '@/schema/card';
import { Hand } from '@/schema/game';

type CardSetKye = 'A' | 'B1' | 'B2';
export const testCards: {
  [hand in Hand]: {
    [key in CardSetKye]: CardSet<Card>;
  };
} = {
  [Hand.RoyalStraightFlush]: {
    A: [
      { suit: 'Spades', number: 10 },
      { suit: 'Spades', number: 11 },
      { suit: 'Spades', number: 12 },
      { suit: 'Spades', number: 13 },
      { suit: 'Spades', number: 1 },
    ], // ロイヤルストレートフラッシュ(10♠ J♠ Q♠ K♠ A♠)
    B1: [
      { suit: 'Clubs', number: 10 },
      { suit: 'Clubs', number: 11 },
      { suit: 'Clubs', number: 12 },
      { suit: 'Clubs', number: 13 },
      { suit: 'Clubs', number: 1 },
    ], // ロイヤルストレートフラッシュ(10♣ J♣ Q♣ K♣ A♣)
    B2: [], // 使用しない
  },
  [Hand.StraightFlush]: {
    A: [
      { suit: 'Spades', number: 9 },
      { suit: 'Spades', number: 10 },
      { suit: 'Spades', number: 11 },
      { suit: 'Spades', number: 12 },
      { suit: 'Spades', number: 13 },
    ], // ストレートフラッシュ(9♠ 10♠ J♠ Q♠ K♠)
    B1: [
      { suit: 'Spades', number: 4 },
      { suit: 'Spades', number: 5 },
      { suit: 'Spades', number: 6 },
      { suit: 'Spades', number: 7 },
      { suit: 'Spades', number: 8 },
    ], // ストレートフラッシュ(4♠ 5♠ 6♠ 7♠ 8♠)
    B2: [
      { suit: 'Diamonds', number: 9 },
      { suit: 'Diamonds', number: 10 },
      { suit: 'Diamonds', number: 11 },
      { suit: 'Diamonds', number: 12 },
      { suit: 'Diamonds', number: 13 },
    ], // ストレートフラッシュ(9♦ 10♦ J♦ Q♦ K♦)
  },
  [Hand.FourOfAKind]: {
    A: [
      { suit: 'Clubs', number: 4 },
      { suit: 'Spades', number: 13 },
      { suit: 'Hearts', number: 13 },
      { suit: 'Diamonds', number: 13 },
      { suit: 'Clubs', number: 13 },
    ], // フォーカード(4♣ K♠ K♥ K♦ K♣)
    B1: [
      { suit: 'Spades', number: 7 },
      { suit: 'Hearts', number: 7 },
      { suit: 'Diamonds', number: 7 },
      { suit: 'Clubs', number: 7 },
      { suit: 'Clubs', number: 12 },
    ], // フォーカード(7♠ 7♦ 7♥ 7♣ Q♣)
    B2: [], // 使用しない
  },
  [Hand.FullHouse]: {
    A: [
      { suit: 'Spades', number: 12 },
      { suit: 'Hearts', number: 12 },
      { suit: 'Spades', number: 13 },
      { suit: 'Hearts', number: 13 },
      { suit: 'Diamonds', number: 13 },
    ], // フルハウス(Q♠ Q♥ K♠ K♠ K♦)
    B1: [
      { suit: 'Spades', number: 2 },
      { suit: 'Hearts', number: 2 },
      { suit: 'Spades', number: 3 },
      { suit: 'Hearts', number: 3 },
      { suit: 'Diamonds', number: 3 },
    ], // フルハウス(2♠ 2♦ 3♦ 3♠ 3♥)
    B2: [], // 使用しない
  },
  [Hand.Flush]: {
    A: [
      { suit: 'Spades', number: 2 },
      { suit: 'Spades', number: 6 },
      { suit: 'Spades', number: 9 },
      { suit: 'Spades', number: 13 },
      { suit: 'Spades', number: 1 },
    ], // フラッシュ(2♠ 6♠ 9♠ K♠ A♠)
    B1: [
      { suit: 'Clubs', number: 3 },
      { suit: 'Clubs', number: 4 },
      { suit: 'Clubs', number: 5 },
      { suit: 'Clubs', number: 7 },
      { suit: 'Clubs', number: 8 },
    ], // フラッシュ(3♠ 4♠ 5♠ 7♠ 8♠)
    B2: [
      { suit: 'Clubs', number: 2 },
      { suit: 'Clubs', number: 6 },
      { suit: 'Clubs', number: 9 },
      { suit: 'Clubs', number: 13 },
      { suit: 'Clubs', number: 1 },
    ], // フラッシュ(2♣ 6♣ 9♣ K♣ A♣)
  },
  [Hand.Straight]: {
    A: [
      { suit: 'Spades', number: 10 },
      { suit: 'Spades', number: 11 },
      { suit: 'Spades', number: 12 },
      { suit: 'Spades', number: 13 },
      { suit: 'Hearts', number: 1 },
    ], // ストーレート(10♠ J♠ Q♠ K♠ A♥)
    B1: [
      { suit: 'Spades', number: 4 },
      { suit: 'Spades', number: 5 },
      { suit: 'Spades', number: 6 },
      { suit: 'Spades', number: 7 },
      { suit: 'Hearts', number: 8 },
    ], // フラッシュ(4♠ 5♦ 6♥ 7♣ 8♦)
    B2: [
      { suit: 'Diamonds', number: 10 },
      { suit: 'Diamonds', number: 11 },
      { suit: 'Diamonds', number: 12 },
      { suit: 'Diamonds', number: 13 },
      { suit: 'Clubs', number: 1 },
    ], // フラッシュ(10♦ J♦ Q♦ K♦ A♣)
  },
  [Hand.ThreeOfAKind]: {
    A: [
      { suit: 'Spades', number: 5 },
      { suit: 'Spades', number: 6 },
      { suit: 'Spades', number: 8 },
      { suit: 'Hearts', number: 8 },
      { suit: 'Diamonds', number: 8 },
    ], // スリーカード(5♠ 6♠ 8♠ 8♥ 8♦)
    B1: [
      { suit: 'Spades', number: 5 },
      { suit: 'Spades', number: 6 },
      { suit: 'Spades', number: 7 },
      { suit: 'Hearts', number: 7 },
      { suit: 'Diamonds', number: 7 },
    ], // スリーカード(3♠ 4♠ 7♠ 7♥ 7♦)
    B2: [], // 使用しない
  },
  [Hand.TwoPair]: {
    A: [
      { suit: 'Spades', number: 3 },
      { suit: 'Spades', number: 5 },
      { suit: 'Hearts', number: 5 },
      { suit: 'Spades', number: 7 },
      { suit: 'Hearts', number: 7 },
    ], // ツーペア(3♠ 5♠ 5♥ 7♠ 7♥)
    B1: [
      { suit: 'Spades', number: 2 },
      { suit: 'Spades', number: 4 },
      { suit: 'Hearts', number: 4 },
      { suit: 'Spades', number: 6 },
      { suit: 'Hearts', number: 6 },
    ], // ツーペア(2♠ 4♠ 4♥ 6♠ 6♥)
    B2: [
      { suit: 'Hearts', number: 3 },
      { suit: 'Diamonds', number: 5 },
      { suit: 'Clubs', number: 5 },
      { suit: 'Diamonds', number: 7 },
      { suit: 'Clubs', number: 7 },
    ], // ツーペア(3♥ 5♦ 5♣ 7♦ 7♣)
  },
  [Hand.OnePair]: {
    A: [
      { suit: 'Spades', number: 6 },
      { suit: 'Hearts', number: 6 },
      { suit: 'Spades', number: 7 },
      { suit: 'Diamonds', number: 8 },
      { suit: 'Clubs', number: 9 },
    ], // ワンペア(6♠ 6♥ 7♠ 8♦ 9♣)
    B1: [
      { suit: 'Spades', number: 4 },
      { suit: 'Hearts', number: 4 },
      { suit: 'Spades', number: 9 },
      { suit: 'Diamonds', number: 10 },
      { suit: 'Clubs', number: 11 },
    ], // ワンペア(4♠ 4♥ 9♥ 10♦ J♣)
    B2: [
      { suit: 'Clubs', number: 6 },
      { suit: 'Diamonds', number: 6 },
      { suit: 'Diamonds', number: 7 },
      { suit: 'Clubs', number: 8 },
      { suit: 'Spades', number: 9 },
    ], // ワンペア(6♣ 6♦ 7♦ 8♣ 9♠)
  },
  [Hand.HighCard]: {
    A: [
      { suit: 'Diamonds', number: 2 },
      { suit: 'Spades', number: 4 },
      { suit: 'Clubs', number: 6 },
      { suit: 'Hearts', number: 7 },
      { suit: 'Spades', number: 1 },
    ], // ハイカード(2♦ 4♠ 6♣ 7♥ A♠)
    B1: [
      { suit: 'Clubs', number: 2 },
      { suit: 'Hearts', number: 4 },
      { suit: 'Diamonds', number: 6 },
      { suit: 'Clubs', number: 7 },
      { suit: 'Spades', number: 13 },
    ], // ハイカード(2♣ 4♥ 6♦ 7♣ K♠)
    B2: [
      { suit: 'Clubs', number: 2 },
      { suit: 'Hearts', number: 4 },
      { suit: 'Diamonds', number: 6 },
      { suit: 'Clubs', number: 7 },
      { suit: 'Hearts', number: 1 },
    ], // ハイカード(2♣ 4♥ 6♦ 7♣ A♥)
  },
  [Hand.Drop]: {
    A: [], // 使用しない
    B1: [], // 使用しない
    B2: [], // 使用しない
  },
};
