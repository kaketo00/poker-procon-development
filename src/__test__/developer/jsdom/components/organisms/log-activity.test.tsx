import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';

import { testLogBetAddPoint } from '@/__test__/developer/jsdom/data/log-bet-add-point';
import { testLogBetKeepPoint } from '@/__test__/developer/jsdom/data/log-bet-keep-point';
import { testLogDraw } from '@/__test__/developer/jsdom/data/log-draw';
import { testLogDrawNoExchange } from '@/__test__/developer/jsdom/data/log-draw-no-exchange';
import type { LogActivityType } from '@/components/organisms/log-activity';
import LogActivity from '@/components/organisms/log-activity';
import theme from '@/styles/theme';

const Component = (props: LogActivityType) => (
  <ChakraProvider theme={theme}>
    <LogActivity {...props} />
  </ChakraProvider>
);

describe('GameInformation', () => {
  test('スナップショット/bet - 加算あり', () => {
    const { asFragment } = render(<Component log={testLogBetAddPoint} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/bet - 加算なし', () => {
    const { asFragment } = render(<Component log={testLogBetKeepPoint} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/draw - 交換あり', () => {
    const { asFragment } = render(<Component log={testLogDraw} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/draw - 交換なし', () => {
    const { asFragment } = render(<Component log={testLogDrawNoExchange} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
