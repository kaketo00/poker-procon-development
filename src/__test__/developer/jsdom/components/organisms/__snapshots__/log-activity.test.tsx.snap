// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameInformation スナップショット/bet - 加算あり 1`] = `
<DocumentFragment>
  <div
    class="css-1ec8zan"
  >
    <div
      class="css-1pjqcpo"
    >
      <p
        class="chakra-text css-tb8v7v"
      >
        Pot:
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        600pt
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        →
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        914pt
      </p>
    </div>
    <div
      class="css-1pjqcpo"
    >
      <p
        class="chakra-text css-tb8v7v"
      >
        Min bet point:
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        0pt
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        →
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        314pt
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`GameInformation スナップショット/bet - 加算なし 1`] = `
<DocumentFragment>
  <div
    class="css-1ec8zan"
  >
    <div
      class="css-1pjqcpo"
    >
      <p
        class="chakra-text css-tb8v7v"
      >
        Pot:
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        600pt
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        →
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        600pt
      </p>
    </div>
    <div
      class="css-1pjqcpo"
    >
      <p
        class="chakra-text css-tb8v7v"
      >
        Min bet point:
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        0pt
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        →
      </p>
      <p
        class="chakra-text css-tb8v7v"
      >
        0pt
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`GameInformation スナップショット/draw - 交換あり 1`] = `
<DocumentFragment>
  <div
    class="css-1ec8zan"
  >
    <div
      class="css-fl2o3k"
    >
      <div
        class="css-k3kwl"
      >
        <img
          alt="card"
          class="chakra-image css-1sx3zwc"
          src="/card/Clubs-06.svg"
        />
      </div>
      <p
        class="chakra-text css-1fqgk32"
      >
        ↓
      </p>
      <div
        class="css-k3kwl"
      >
        <img
          alt="card"
          class="chakra-image css-1nhackf"
          src="/card/Hearts-01.svg"
        />
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`GameInformation スナップショット/draw - 交換なし 1`] = `
<DocumentFragment>
  <div
    class="css-1ec8zan"
  >
    <p
      class="chakra-text css-tb8v7v"
    >
      No change.
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
